FontArc support with no_std

I've been toying around with trying to get egui running on no_std recently (with success) and egui makes use of the `FontArc` type from this crate, which is disabled with no_std. As far as I can tell the only reason as to why is the fact that some systems do not support `Arc`. However for my use case, it is available.

Would it be possible to change the std feature gate on this to be `#[cfg(target_has_atomic = "ptr")]` instead? Or alternatively introduce a separate feature gate to manually enable it?

I can open a PR for this, the only thing I ended up having to change was the feature gate and adding a missing use for `Vec`.

ab_glyph does compile, however egui (specifically the epaint crate) relies on the `font_arc` module, which is not available on no_std:
```
   Compiling ab_glyph v0.2.31
   Compiling epaint v0.32.1 (/blah/blah/egui/crates/epaint)
error[E0433]: failed to resolve: could not find `FontArc` in `ab_glyph`
   --> crates/epaint/src/text/fonts.rs:208:32
    |
208 |                 .map(ab_glyph::FontArc::from)
    |                                ^^^^^^^
    |                                |
    |                                could not find `FontArc` in `ab_glyph`
    |                                help: a struct with a similar name exists: `FontVec`
```
Replacing the `#[cfg(feature = "std")]` (i.e. with `#[cfg(target_has_atomic = "ptr")]`) in `lib.rs` and fixing the missing use Vec in `font_arc.rs` continues to compile with no_std and furthermore fixes the problem from within epaint.
The problem isn't specific to any particular target, it happens when compiling for normal desktop linux even. The target I want no_std for is the GameCube, which has support for atomics, hence I'm using a custom target json:
<details>
<summary>powerpc-none-dol.json</summary>

```json
{
    "arch": "powerpc",
    "cpu": "750",
    "data-layout": "E-m:e-p:32:32-Fn32-i64:64-n32",
    "emit-debug-gdb-scripts": false,
    "env": "newlib",
    "executables": true,
    "features": "+hard-float",
    "linker": "powerpc-eabi-gcc",
    "linker-flavor": "gcc",
    "linker-is-gnu": true,
    "llvm-target": "powerpc-none-dol",
    "os": "none",
    "panic-strategy": "abort",
    "pre-link-args": {
        "gcc": [
            "-mogc",
            "-mcpu=750",
            "-meabi",
            "-msdata=eabi",
            "-mhard-float"
        ]
    },
    "post-link-args": {
        "gcc": [
            "-lgcc"
        ]
    },
    "relocation-model": "static",
    "target-c-int-width": "32",
    "target-endian": "big",
    "target-pointer-width": "32",
    "target-family": "unix",
    "target-mcount": "_mcount",
    "vendor": "nintendo"
}
```

</details>

--- a/glyph/CHANGELOG.md
+++ b/glyph/CHANGELOG.md
@@ -1,3 +1,6 @@
+# Unreleased (0.2.32)
+* Add `FontArc` support for no_std targets with cfg(target_has_atomic = "ptr").
+
 # 0.2.31
 * Add "gvar-alloc" feature enabled by default (activates _ttf-parser_ "gvar-alloc" feature).
   Provides full gvar table support.

--- a/glyph/src/font_arc.rs
+++ b/glyph/src/font_arc.rs
@@ -1,5 +1,7 @@
 use crate::{v2, Font, FontRef, FontVec, GlyphId, InvalidFont, Outline};
 use alloc::sync::Arc;
+#[cfg(not(feature = "std"))]
+use alloc::vec::Vec;
 use core::fmt;
 
 /// `Font` implementor that wraps another concrete `Font + 'static` type storing in an `Arc`.

--- a/glyph/src/lib.rs
+++ b/glyph/src/lib.rs
@@ -26,7 +26,7 @@ extern crate alloc;
 mod codepoint_ids;
 mod err;
 mod font;
-#[cfg(feature = "std")]
+#[cfg(target_has_atomic = "ptr")]
 mod font_arc;
 mod glyph;
 #[cfg(all(feature = "libm", not(feature = "std")))]
@@ -37,7 +37,7 @@ mod ttfp;
 #[cfg(feature = "variable-fonts")]
 mod variable;
 
-#[cfg(feature = "std")]
+#[cfg(target_has_atomic = "ptr")]
 pub use crate::font_arc::*;
 #[allow(deprecated)]
 pub use crate::{

--- a/test
+++ b/test
@@ -11,6 +11,8 @@ cargo test --benches
 echo "==> no_std"
 cargo build -p ab_glyph_rasterizer --target thumbv6m-none-eabi --no-default-features --features libm
 cargo build -p ab_glyph --target thumbv6m-none-eabi --no-default-features --features libm
+cargo build -p ab_glyph_rasterizer --no-default-features --features libm
+cargo build -p ab_glyph --no-default-features --features libm
 echo "==> no_std (variable-fonts)"
 cargo build -p ab_glyph --target thumbv6m-none-eabi --no-default-features --features "libm variable-fonts"
 echo "==> check wasm32"

